src/representation_theory/basic.lean

```diff
@@ -107,4 +128,120 @@ theorem char_conj (g : G) (h : G) : (character k G V) (h * g * h⁻¹) = (charac
 
 end character
 
+section linear_hom
+
+variables {k G V W : Type*} [comm_semiring k] [group G] [add_comm_monoid V] [add_comm_monoid W]
+variables [module k V] [distrib_mul_action G V] [smul_comm_class G k V]
+variables [module k W] [distrib_mul_action G W] [smul_comm_class G k W]
+
+/--
+Given two `k`-linear representations `V` and `W` of a group `G`,
+the space are linear maps `V →ₗ[k] W` is itself a representation of `G`.
+-/
+instance  linear_hom_distrib_mul_action : distrib_mul_action G (V →ₗ[k] W) :=
```

Comment:
I'm pretty sure this creates a diamond when `G = units k`

Reply:
But `G` will never be *syntactically* `units k`.

Reply:
Sure it will, in the case where I want to scale the codomain of a linear map by some invertible element of the ring. That is; the other instance _is_ desirable when `G = units k`. I think you're saying that for representation theory, we never have `G = units k` so we don't care about this instance.

My objection is that this diamond makes a mess for people _not_ doing representation theory.